### PR TITLE
Add Sqlite as EF DB provider

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Adds sqlite as DB provider

## Code changes

- **Admin.csproj**: include sqlite migrations project so migrator can trigger migrations for it.
- **startup**: add sqlite as known migrator
- **packages.lock/ EntityFramework.csproj**: install sqlite EF package
- **All others**: Add sqlite as known EF DB provider

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: dev/.gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 .data
 secrets.json
+db
```
Comment (kspearrin) [inline dev/.gitignore:3]: What is this for?

File: src/Infrastructure.EntityFramework/EntityFrameworkServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -31,6 +31,10 @@ public static class EntityFrameworkServiceCollectionExtensions
                 options.UseMySql(connectionString, ServerVersion.AutoDetect(connectionString),
                     b => b.MigrationsAssembly("MySqlMigrations"));
             }
+            else if (provider == SupportedDatabaseProviders.Sqlite)
+            {
+                options.UseSqlite(connectionString);
```
Comment (kspearrin) [inline src/Infrastructure.EntityFramework/EntityFrameworkServiceCollectionExtensions.cs:36]: Do we not need the `MigrationsAssembly` definition here?

**Review by kspearrin — COMMENTED** [review]
Just a couple questions. I think Justin has some ideas to clean up and get rid of all of these migrations projects in the future.

File: dev/.gitignore (hunk 1 above)
Comment (MGibson1) [reply dev/.gitignore:3]: to allow people to have a dev environment with a `.sqlite` db.

I've added a connection string to `secrets.json.example` with a connection string point to this directory.

File: src/SharedWeb/Utilities/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -66,6 +66,10 @@ public static SupportedDatabaseProviders AddDatabaseRepositories(this IServiceCo
                     provider = SupportedDatabaseProviders.MySql;
                     connectionString = globalSettings.MySql.ConnectionString;
                     break;
+                case "sqlite":
```
Comment (kspearrin) [inline src/SharedWeb/Utilities/ServiceCollectionExtensions.cs:69]: maybe add `sqllite` for misspellings? Not sure if that is a thing.

**Comment by MGibson1** [comment]
Need to figure out the testing and self host build issues, but should be ready for another code review

**Pull request merged**